public/app/features/explore/state/time.ts

```diff
@@ -160,6 +168,39 @@ export function zoomOut(scale: number): ThunkResult<void> {
   });
 }
 
+export function copyTimeRangeToClipboard(): ThunkResult<void> {
+  return (dispatch, getState) => {
+    const range = getState().explore.panes[Object.keys(getState().explore.panes)[0]]!.range.raw;
+    navigator.clipboard.writeText(JSON.stringify(range));
+
+    appEvents.emit(AppEvents.alertSuccess, ['Time range copied to clipboard']);
+  };
+}
+
+export function pasteTimeRangeFromClipboard(): ThunkResult<void> {
+  return async (dispatch, getState) => {
+    const raw = await navigator.clipboard.readText();
```

Comment:
the logic for reading the text, checking if it's valid, and returning a valid range or error can be extracted into a separate function that can be shared between explore and dashboards, but wasn't sure if we should do that. if we decide to do it, i'm sure where to put that function.

Reply:
Maybe here: https://github.com/grafana/grafana/blob/main/public/app/core/utils/timePicker.ts?